src/Picasso_ParticleInit.hpp

```diff
@@ -37,16 +37,20 @@ struct InitRandom
 template <class CreationView, class ParticleAoSoA, class ExecutionSpace>
 void filterEmpties( const ExecutionSpace& exec_space,
                     const int local_num_create,
+                    const int previous_num_particles,
                     const CreationView& particle_created,
-                    ParticleAoSoA& particles )
+                    ParticleAoSoA& particles, const bool shrink_to_fit = true )
 {
     using memory_space = typename CreationView::memory_space;
 
     // Determine the empty particle positions in the compaction zone.
     int num_particles = particles.size();
+    // int max_local_num_create = previous_num_particles + local_num_create;
     Kokkos::View<int*, memory_space> empties(
         Kokkos::ViewAllocateWithoutInitializing( "empties" ),
-        std::min( num_particles - local_num_create, local_num_create ) );
+        std::min( num_particles - previous_num_particles - local_num_create,
+                  local_num_create ) );
+
     Kokkos::parallel_scan(
         "Picasso::ParticleInit::FindEmpty",
         Kokkos::RangePolicy<ExecutionSpace>( exec_space, 0, local_num_create ),
```

Comment:
Should this range policy only be up to `local_num_create`? Shouldn't it be `particle_created.size()`, which is on line 157.

Reply:
No, we only need the new index for the number that we create